lib/vecmat.gi

```diff
@@ -1586,10 +1595,16 @@ local sf, rep, ind, ind2, row, i,big,l;
     rep:=IsPlistRep;
   fi;
 
+  if IsPlistRep(matrix) and not ForAll(matrix,IsPlistRep) then
+    nr:=Length(matrix);
+  else
+    nr:=NrRows(matrix);
+  fi;
+
```

Comment:
Why is this necessary? `NrRows` should work for "old-style" matrices, too

Reply:
Did not realize that -- chaned.

Reply:
Actually...
It is needed, because the input could be a mixed format list, for which `NrRows` will not work.

Reply:
Could you elaborate on that a bit? What is a "mixed format list" and where does it occur?

Reply:
While merged already, a brief explanation for later:

The function is `ImmutableMatrix`. It takes somethig that could be construed as a list of vectors, and converts it into a compact format matrix (with the implicit claim that this representation will be efficient to work with).
This could be a list of vectors of different storage/mutability/base ring status (and indeed, this can happen when calling from the MeatAxe).